connection.go

```diff
@@ -88,20 +99,31 @@ func (p *Pool) ExecQuery(query string) ([]byte, error) {
 }
 
 func (p *Pool) Exec(req *Request) ([]byte, error) {
-	conn, err := p.Get()
-	if err != nil {
-		return nil, err
-	}
+	conn := p.Get()
 	requestMessage, err := GraphSONSerializer(req)
 	if err != nil {
+		p.logger.Log("error while serialising request", err)
 		return nil, err
 	}
-	// Open a TCP connection
+start:
 	if err := websocket.Message.Send(conn.ws, requestMessage); err != nil {
+		p.logger.Log("error while sending message", err)
 		return nil, err
 	}
-
 	data, err := conn.ReadResponse()
+	if err != nil {
+		p.logger.Log("error while reading response", err)
```

Comment:
In logger, the way it works is like key, value, so first argument is the key, second is the value, third is key, fourth is value and so on. Key cannot have spaces in it.

Reply:
We should log this error only in the else case, wherein the error is not io.EOF(), as that one we are already handling.